### DISCUSSION TITLE
How to integrate with the mouse menu?

### DISCUSSION THREAD

**Discussion by airtonix** [nvim-neo-tree/neo-tree.nvim#665] [Q&A]
Mouse using heretic here 👋🏻 😄 

![firefox_bHjgOyjpoj](https://user-images.githubusercontent.com/61225/209462968-223c3436-6840-4255-b5d5-6a88b57d8ae5.gif)

Would love to know how to get all the awesome neo-tree things into this mouse menu?

**Comment by cseickel** [comment]
Is that menu even configurable?

**Reply by airtonix** [reply]
It seems like it might be yeah: 

- https://www.reddit.com/r/neovim/comments/xz9940/today_i_learned_that_neovim_has_a_right_click_menu/
![](https://user-images.githubusercontent.com/4246425/60061310-4180f580-9727-11e9-980d-ae8325f34a7d.JPG)

```
:h default-mouse
:h popup-menu
```
https://www.reddit.com/r/neovim/comments/xz9940/comment/irlq9w6/?utm_source=reddit&utm_medium=web2x&context=3


This person made a plugin, which does result in the same kind of UX, but it's completely different implementation to what I'm referring to: 

- https://github.com/kvngvikram/rightclick-macros

**Reply by miversen33** [reply]
This smells like a feature request. @airtonix could you open an issue with some details on what you would like this to look like?

Stuff that would be helpful, what do you want the mouse menu to contain by default when you right click on an item in neo-tree?
I don't know what the release schedule looks like, but its probably worth getting this tagged properly so it can be prioritized :)

**Reply by airtonix** [reply]
@miversen33 Raised #772 👍🏻
